tests/unit/openpype/modules/sync_server/test_module_api.py

```diff
@@ -0,0 +1,46 @@
+"""Test file for Sync Server, tests API methods, currently for integrate_new
+
+    File:
+        creates temporary directory and downloads .zip file from GDrive
+        unzips .zip file
+        uses content of .zip file (MongoDB's dumps) to import to new databases
+        with use of 'monkeypatch_session' modifies required env vars
+            temporarily
+        runs battery of tests checking that site operation for Sync Server
+            module are working
+        removes temporary folder
+        removes temporary databases (?)
+"""
+import pytest
+
+from tests.lib.testing_classes import ModuleUnitTest
+
+
+class TestModuleApi(ModuleUnitTest):
+
+    REPRESENTATION_ID = "60e578d0c987036c6a7b741d"
+
+    TEST_FILES = [("1eCwPljuJeOI8A3aisfOIBKKjcmIycTEt",
+                   "test_site_operations.zip", '')]
+
+    @pytest.fixture(scope="module")
+    def setup_sync_server_module(self, dbcon):
+        """Get sync_server_module from ModulesManager"""
+        from openpype.modules import ModulesManager
+
+        manager = ModulesManager()
+        sync_server = manager.modules_by_name["sync_server"]
+        yield sync_server
+
+    def test_get_alt_site_pairs(self, setup_sync_server_module):
+        conf_sites = {'SFTP': {"alternative_sites": ["studio"]},
+                      "studio2": {"alternative_sites": ["studio"]}}
+
+        ret = setup_sync_server_module._get_alt_site_pairs(conf_sites)
+        expected = {"SFTP": ["studio", "studio2"],
+                    "studio": ["SFTP", "studio2"],
+                    "studio2": ["studio", "SFTP"]}
+        assert ret == expected, "Not matching result"
```

Comment:
I feel like just one level of alternate sites that matches doesn't accurately represent whether we tackle 'cycles' well enough and whether the recursion works across a depth of one.

Should we write a test for case like this:
```
        conf_sites = {"A": {"alternative_sites": ["C"]},
                      "B": {"alternative_sites": ["C"]},
                      "C": {"alternative_sites": ["D"]},
                      "D": {"alternative_sites": ["A"]},
                      "F": {"alternative_sites": ["A"]},
                      }
```
This tackles A -> B -> C -> D

Also - your test only passes if the order of the output is actually like you have it defined here - however. Is that order really relevant for the pass to test? Like, do we absolutely require them to be ordered a certain way in the lists? I just wasn't sure if that had any value - if it doesn't have value we might want to use `set` for the alternatives?

Reply:
Good idea, added more complex unit test.